jest/setup.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
```

Comment:
Is this file based on some example? What's the source of truth? When this breaks or isn't sufficient with future RN versions, how will the next person fix it?

Reply:
It's loosely based on https://github.com/facebook/react-native/blob/master/jest/setup.js. I've had to change parts though to make it work with this. If this breaks, someone will need to fix it, I don't see any other way. It sucks, but then so does the need to mock everything in RN.

Reply:
Are you saying that any RN apps that want to use Jest also copy paste this? Or is there another way?

Reply:
You can use the react-native preset, which pulls these exact same files in. The only issue is that the preset mocks View and Text, and we don't want them to be mocked, we want the real version so we can compare the "inlined" View and Text (RCTView and RCTText/RCTVirtualText). I asked the RN folks and they guided me to these files and said I should fork them to do what I want with Jest.

Reply:
Why can't we use RN preset + `jest.unmock` to undo the mocks?

Reply:
I'll try that out, I never even knew there was an unmock! Likely next week though, unless you wanted to pick this up? Seems you know more about Jest than I do.